nvme.c

```diff
@@ -7289,33 +7195,30 @@ static int submit_io(int opcode, char *command, const char *desc, int argc, char
 	}
 
 	if (strlen(cfg.metadata)) {
+		close(mfd);
 		mfd = open(cfg.metadata, flags, mode);
 		if (mfd < 0) {
 			nvme_show_perror(cfg.metadata);
-			err = -EINVAL;
-			goto close_dfd;
+			return -EINVAL;
 		}
 	}
 
 	if (!cfg.data_size) {
 		nvme_show_error("data size not provided");
-		err = -EINVAL;
-		goto close_mfd;
+		return -EINVAL;
 	}
 
 	ns = nvme_alloc(sizeof(*ns));
-	if (!ns) {
-		err = -ENOMEM;
-		goto close_mfd;
-	}
+	if (!ns)
+		return -ENOMEM;
 
 	err = nvme_cli_identify_ns(dev, cfg.namespace_id, ns);
 	if (err > 0) {
 		nvme_show_status(err);
-		goto close_mfd;
+		return -err;
```

Comment:
We should properly just return `-EINVAL` because all error codes inside nvme are negative values. That was also the reason why I switched it to `-err`.

Reply:
After looking at the rest, we have pretty much half of  error values positive and negative in nvme.c. So let's keep it. Something for later.